.github/workflows/python-app.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+env:
+  NODE_VERSION: '10.x'                # set this to the node version to use
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: npm install
+      run: |
+        sudo npm install
+        sudo npm install -g jscpd@3.2.1
+    - name: Static Analysis
+      run: |
+        pylint guardrails/ test/
+        jscpd --min-tokens 20 --reporters "json" --mode "strict" --format "python" -o . .
+        python3 test_resource/jscpd_parser.py --j 10 #Allowed duplication is 10% with 20 token size
+
+    - name: Cyclomatic complexity
+      run: |
+        python3 -m lizard guardrails -X> CC.xml
+        python3 test_resource/cyclo_gate.py --c 10 #Allowed cyclomatic complexity is less than 10
+    - name: Dead Code
+      run: |
+        python3 -m vulture guardrails
+    - name: Tests & Coverage
+      run: |
+        pytest test --cov-config=.coveragerc --cov=guardrails
+        coverage report --fail-under=90 #Coverage should be above 90%
+        codecov
+    - name: Mutation Tests
+      run: |
+        mutmut run > mutmut.log || true # Checking mutation on main file as the travis max execution time is overshooting
+        mutmut junitxml --suspicious-policy=ignore --untested-policy=ignore > mutmut.xml
+        python3 test_resource/mutmut_parse.py --m 20 #Mutation (failure+disables+error) allowed 20% of total mutants
```

Comment:
If tomorrow we increased the % of killed mutants, would we have to change this yml file to reduce the allowed percentage as well?

Reply:
No, this 20%, is max allowed escape mutants.